rfcs/20230807-Unify-primitive-and-graph-verbose/README.md

```diff
@@ -0,0 +1,139 @@
+
+To better unify the user interface with primitive verbose, this proposal is to
+merge `ONEDNN_GRAPH_VERBOSE` into `ONEDNN_VERBOSE`:
+
+- `ONEDNN_GRAPH_VERBOSE`  will be deprecated and graph verbose can be controlled
+  via `ONEDNN_VERBOSE`
+- When `ONEDNN_VERBOSE=1`, verbose of both primitive and compiled partition exec
+  profiling will be printed
+- When `ONEDNN_VERBOSE=2`, verbose of both primitive creation/execution,
+  partition compilation and compiled partition execution profiling will be
+  printed
+- When `ONEDNN_VERBOSE=error/check/debug`, verbose of error/check/debug info
+  from both primitive and graph implementation will be printed
+- Specially, `profile/profile_create/profile_exec` won’t be changed and
+  `graph:profile/graph:profile_compile/ graph:profile_exec` will be added to
+  support print profile info from graph API only. These can be combined with the
+  current string-based value and give more flexibility to users when doing some
+  profiling work.
+
+To achieve this, `get_verbose()` and `verbose_t` need to be extended and
+`get_graph_verbose()` will be removed.
+
+### Unified Verbose Format
+
+To better unify the msg log with primitive log, this proposal is to align with
+the marker `onednn_verbose` and add a new column `primitive/graph` to indicate
+where the verbose came from. Take execution profiling as an example, we can see
+the first line came from primitive and the second came from graph API:
+
+```bash
+# primitive execution verbose
+onednn_verbose,primitive,exec,cpu,convolution,brgconv_1x1:avx512_core,forward_inference,src_f32::blocked:acdb::f0 wei_f32:a:blocked:Acdb64a::f0 bia_f32:a:blocked:a::f0 dst_f32::blocked:acdb::f0,attr-scratchpad:user attr-post-ops:eltwise_relu ,alg:convolution_direct,mb2_ic32oc64_ih112oh112kh1sh1dh0ph0_iw112ow112kw1sw1dw0pw0,1.15503
+
+# compiled partition execution verbose
+onednn_verbose,graph,exec,cpu,100006,convolution_post_ops,CONV_0;ELTWISE_1,data:NCX; filter:OIX;;,in0_f32:0:strided:undef:2x32x112x112:401408s12544s112s1 in1_f32:1:strided:constant:64x32x1x1:32s1s1s1 in2_f32:2:strided:constant:64:1 out0_f32:4:strided:undef:2x64x112x112:802816s12544s112s1,fpm:strict,dnnl_backend,5.44092
+```
+
+Also, like the primitive API, the graph API also supports some checking info and
+the msg log will be shown as:
+
+```bash
+# primitive check for invalid u8f32f32 matmul primitive
+onednn_verbose,primitive,create:check,matmul,invalid datatype for accumulation,src/common/matmul.cpp:188
+
+# graph check for invalid u8f32f32 matmul op
+onednn_verbose,graph,graph:check,add_op,MatMul,given data type for input0 is u8 v.s. expected {f16,bf16,f32},src/graph/interface/op_schema.cpp:250
+```
+
+To achieve this, graph related msg need to be added in `verbose_msg.hpp` and
+`VPROF/VCHECK/VINFO` need to be adjusted. Duplicated macros like
+`VPROFGRAPH/VCHECKGRAPH/VINFOGRAPH` can be removed. Also, minor changes may
+happen in `verbose_converter` side to only convert verbose from primitive API.
```

Comment:
I would suggest to have a dedicate section in this RFC to describe the potential impacts to verbose_converter.

Reply:
added